### PR TITLE
impl `HasPayload`, `Request` and `Requester` for `either::Either`

### PR DESCRIPTION
**Pull request by WaffleLapkin**
This allows to use different adaptors depending on runtime context.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -75,7 +76,10 @@ cache_me = []
 # AutoSend bot adaptor
 auto_send = []
 
-full = ["throttle", "cache_me", "auto_send"]
+# Erased bot adaptor
+erased = []
```
Comment (Hirrolot) [inline Cargo.toml:80]: Please document this feature in `lib.rs`.

File: Cargo.toml [hunk 2]
```diff
@@ -51,6 +51,7 @@ thiserror = "1.0.20"
 once_cell = "1.5.0"
 never = "0.1.0"
 chrono = "0.4.19"
+either = "1.6.1"
```
Comment (Hirrolot) [inline Cargo.toml:54]: Maybe just `1.6` would suffice?

**Review by Hirrolot — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (WaffleLapkin) [reply Cargo.toml:54]: I think it would, but we use precise versions everywhere in the `Cargo.toml`, so maybe it's a topic for a different PR?

Comment (Hirrolot) [reply Cargo.toml:54]: It seems that libraries should use `x.y` instead of `x.y.z` except for the real need for bug fixes.
Yes, a topic for another PR.

**Review by Hirrolot — APPROVED** [review]

**Pull request merged**